server/services/prismic.js

```diff
@@ -259,23 +254,146 @@ function convertPrismicResultsToPaginatedResults(prismicResults: Object): (resul
+    const start = london(currEvent.start);
+    const end = london(currEvent.end);
+    if (end.isSame(start, 'month')) {
+      const year = start.format('YYYY');
+      const month = start.format('MMMM');
+      if (!acc[year]) {
+        acc[year] = {};
+      }
+      if (!acc[year][month]) {
+        acc[year][month] = [];
+      }
+      acc[year][month].push(currEvent);
+    } else {
+      while ((end.isAfter(start) || end.isSame(start, 'month')) && start.isBefore(london().add(3, 'month'))) {
+        const year = start.format('YYYY');
+        const month = start.format('MMMM');
+        if (!acc[year]) {
+          acc[year] = {};
+        }
+        if (!acc[year][month]) {
+          acc[year][month] = [];
+        }
+        acc[year][month].push(currEvent);
+        start.add(1, 'month');
+      }
+    }
+    return acc;
+  }, {});
+}
+
+export async function getExhibitionAndEventPromos(queryDates) {
+  const todaysDate = london();
+  const dateRange = queryDates && queryDates.split('|');
+  const fromDate = dateRange && dateRange[0] ? dateRange[0] : todaysDate.format('YYYY-MM-DD');
+  const toDate = dateRange && dateRange[1] ? dateRange[1] : todaysDate.format('YYYY-MM-DD');
+
+  const prismic = await getPrismicApi();
+  const allExhibitionsAndEvents = await prismic.query([
+    Prismic.Predicates.any('document.type', ['exhibitions', 'events'])
+  ]);
+  const exhibitionPromos = createExhibitionPromos(allExhibitionsAndEvents.results.filter(e => e.type === 'exhibitions'));
+  const permanentExhibitionPromos = exhibitionPromos.filter(e => !e.end);
+  const temporaryExhibitionPromos = filterPromosByDate(exhibitionPromos.filter(e => e.end), fromDate, toDate);
+  const eventPromos = filterPromosByDate(createEventPromos(allExhibitionsAndEvents.results.filter(e => e.type === 'events')), fromDate, toDate);
+  const eventPromosGroupedByMonth = groupPromosByMonth(eventPromos);
+  const monthControls = Object.keys(eventPromosGroupedByMonth).reduce((acc, year) => {
+    Object.keys(eventPromosGroupedByMonth[year]).forEach((month) => {
+      const controlObject = {
+        id: `${year}-${month}`,
+        title: month,
+        url: `#${month}-${year}`
+      };
+      acc.push(controlObject);
+    });
+    return acc;
+  }, []);
+
+  const dates = {
+    today: todaysDate.format('YYYY-MM-DD'),
+    weekend: [getWeekendFromDate(todaysDate).format('YYYY-MM-DD'), getWeekendToDate(todaysDate).format('YYYY-MM-DD')],
+    all: [todaysDate.format('YYYY-MM-DD'), london().add(3, 'month').format('YYYY-MM-DD')],
+    queriedDates: dateRange
+  };
+  const active = getActiveState(todaysDate, [fromDate, toDate]);
+  return {
+    active,
+    dates,
+    permanentExhibitionPromos,
+    temporaryExhibitionPromos,
+    eventPromos,
+    eventPromosGroupedByMonth,
+    monthControls
+  };
+}
+
+function getWeekendFromDate(today) {
+  const todayInteger = today.day(); // day() return Sun as 0, Sat as 6
```

Comment:
would `return london(today).day(todayInteger ? -2 : 5);` work?

Reply:
It would, but I find it easier to follow how it is